File: 1_two_sum.py
nums = [2, 7, 11, 15]
target = 9

d = {}
ans = []
for i, n in enumerate(nums):
    m = target - n
    if m in d:
        ans = [d[m], i]
        break
    else:
        d[n] = i
print(ans)

File: 1_two_sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [[Int: Int]] = []
        for i in 0...nums.count - 1 {
            let n = nums[i]
            let m = target - n
            let x = dict.first {
                $0.first?.key == m
            }
            if x != nil {
                return[x!.first!.value, i]
            } else {
                dict.append([n:i])
            }
        }
        return [0, 0]
    }
}
